src/middleware/datasetOverview.middleware.js

```
@@ -90,20 +93,83 @@ const fetchSources = fetchMany({
+ * @returns {function} A middleware function that sets notices based on the source data.
+ */
+export const setNoticesFromSourceKey = (sourceKey) => (req, res, next) => {
+  const { dataset } = req.params
+  const source = req[sourceKey]
+
+  const deadlineObj = requiredDatasets.find(deadline => deadline.dataset === dataset)
+
+  if (deadlineObj) {
+    const noticePeriod = typeof deadlineObj.noticePeriod === 'string' ? parseInt(deadlineObj.noticePeriod, 10) : deadlineObj.noticePeriod
+
+    if (isNaN(noticePeriod) || typeof noticePeriod !== 'number') {
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)